services/sonar/sonar-base.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const Joi = require('joi')
+const serverSecrets = require('../../lib/server-secrets')
+const { BaseJsonService } = require('..')
+const { isLegacyVersion } = require('./sonar-helpers')
+
+const schema = Joi.object({
+  component: Joi.object({
+    measures: Joi.array()
+      .items(
+        Joi.object({
+          metric: Joi.string(),
+          value: Joi.alternatives(
+            Joi.number().min(0),
+            Joi.allow('OK', 'ERROR')
+          ).required(),
+        }).required()
+      )
+      .required(),
+  }).required(),
+}).required()
+
+const legacyApiSchema = Joi.array()
+  .items(
+    Joi.object({
+      msr: Joi.array()
+        .items(
+          Joi.object({
+            key: Joi.string(),
```

Comment:
Also here. Should it be `.required()` ?

Reply:
Have made this required, and no issues 👍